homeassistant/components/integration/sensor.py

```diff
@@ -433,38 +428,72 @@ async def async_added_to_hass(self) -> None:
             source_state = self.hass.states.get(self._sensor_source_id)
             self._schedule_max_sub_interval_exceeded_if_state_is_numeric(source_state)
             self.async_on_remove(self._cancel_max_sub_interval_exceeded_callback)
-            handle_state_change = self._integrate_on_state_change_and_max_sub_interval
+            handle_state_change = self._integrate_on_state_change_with_max_sub_interval
+            handle_state_report = self._integrate_on_state_report_with_max_sub_interval
         else:
             handle_state_change = self._integrate_on_state_change_callback
+            handle_state_report = self._integrate_on_state_report_callback
 
         if (
             state := self.hass.states.get(self._source_entity)
         ) and state.state != STATE_UNAVAILABLE:
             self._derive_and_set_attributes_from_state(state)
 
         self.async_on_remove(
-            async_track_state_change_event(
-                self.hass,
-                [self._sensor_source_id],
+            self.hass.bus.async_listen(
+                EVENT_STATE_CHANGED,
                 handle_state_change,
+                event_filter=callback(
+                    lambda event_data: event_data["entity_id"] == self._sensor_source_id
+                ),
+                run_immediately=True,
+            )
+        )
+        self.async_on_remove(
+            self.hass.bus.async_listen(
+                EVENT_STATE_REPORTED,
+                handle_state_report,
+                event_filter=callback(
+                    lambda event_data: event_data["entity_id"] == self._sensor_source_id
+                ),
+                run_immediately=True,
```

Comment:
I'm getting profiles from beta users this lambda being called 1M/min

I think we need to switch the first one back to `async_track_state_change_event` and than make an `async_track_state_change_reported` so we can dispatch based on entity id since we now have `2 * X * integration entities` top level listeners with this change

Reply:
Small snapshot of the lambdas being called.
<img width="485" alt="Screenshot 2024-06-26 at 5 16 48 PM" src="https://github.com/home-assistant/core/assets/663432/bc07e86d-7690-4362-bf7d-ec795942175a">